tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -252,7 +250,23 @@ public static synchronized SpanObserver unsubscribe(String name) {
     }
 
     private static void computeObserversList() {
-        observersList = ImmutableList.copyOf(observers.values());
+        Consumer<Span> newCompositeObserver = span -> { };
```

Comment:
If we're micro-optimising, why do we tolerate calling the empty observer first and calling `andThen` on it?

Reply:
it's just a method invocation which is free compared to anything that does allocation (list iterator, for example).

Reply:
I can get rid of it, but I don't think it's worth the complexity since we won't be able to measure a difference.